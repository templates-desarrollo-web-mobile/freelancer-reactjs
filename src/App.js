import React from "react";
import Menu from "./components/Top/Menu/Menu";
import Footer from "./components/Bottom/Footer";
import FooterInfo from "./components/Bottom/FooterInfo";
import Contact from "./components/Bottom/Contact";
import About from "./components/Middle/About";
import Portfolio from "./components/Middle/Portfolio";
import Master from "./components/Top/Master";
import Modal from "./components/Others/Modal";
import PropTypes from "prop-types";

const App = ({ portfolioItems }) => (
  <div>
    <Menu />
    <Master />
    <Portfolio items={portfolioItems} />
    <About />
    <Contact />
    <FooterInfo />
    <Footer />
    {portfolioItems.map((icon, index) => (
      <Modal key={icon.id} icon={icon} itemNumber={index + 1} />
    ))}
  </div>
);
export default App;

// Para obligar pasar el argumento para la funci√≥n
App.propTypes = {
  portfolioItems: PropTypes.array.isRequired,
};
