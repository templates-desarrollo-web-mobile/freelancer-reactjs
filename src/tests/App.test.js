import { shallow } from "enzyme";
import App from "../App";
describe('Pruebas en "App"', () => {
  const portfolioItems = [
    { id: "cabin", name: "Log Cabin" },
    { id: "cake", name: "Tasty Cake" },
  ];
  const wrapper = shallow(<App portfolioItems={portfolioItems} />);
  test("Debe de mostrar correctamente el componente", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("Debe mostrar todos los componentes asociados", () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("Menu").length).toBe(1);
    expect(wrapper.find("Portfolio").length).toBe(1);
    expect(wrapper.find("About").length).toBe(1);
    expect(wrapper.find("Contact").length).toBe(1);
    expect(wrapper.find("FooterInfo").length).toBe(1);
    expect(wrapper.find("Footer").length).toBe(1);
    expect(wrapper.find("Modal").length).toBe(2);
  });
});
