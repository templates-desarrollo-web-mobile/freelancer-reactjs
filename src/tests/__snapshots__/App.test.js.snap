// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en "App" Debe de mostrar correctamente el componente 1`] = `
<div>
  <Menu />
  <Master />
  <Portfolio
    items={
      Array [
        Object {
          "id": "cabin",
          "name": "Log Cabin",
        },
        Object {
          "id": "cake",
          "name": "Tasty Cake",
        },
      ]
    }
  />
  <About />
  <Contact />
  <FooterInfo />
  <Footer />
  <Modal
    icon={
      Object {
        "id": "cabin",
        "name": "Log Cabin",
      }
    }
    itemNumber={1}
    key="cabin"
  />
  <Modal
    icon={
      Object {
        "id": "cake",
        "name": "Tasty Cake",
      }
    }
    itemNumber={2}
    key="cake"
  />
</div>
`;

exports[`Pruebas en "App" Debe mostrar todos los componentes asociados 1`] = `
<div>
  <Menu />
  <Master />
  <Portfolio
    items={
      Array [
        Object {
          "id": "cabin",
          "name": "Log Cabin",
        },
        Object {
          "id": "cake",
          "name": "Tasty Cake",
        },
      ]
    }
  />
  <About />
  <Contact />
  <FooterInfo />
  <Footer />
  <Modal
    icon={
      Object {
        "id": "cabin",
        "name": "Log Cabin",
      }
    }
    itemNumber={1}
    key="cabin"
  />
  <Modal
    icon={
      Object {
        "id": "cake",
        "name": "Tasty Cake",
      }
    }
    itemNumber={2}
    key="cake"
  />
</div>
`;
